File: BancoPython/Employee.py
class Employee:
    
    __name = ""
    __age = 0
    __salary = 0.0
    cuentas = [] 

    def __init__(self, name, age, salary ):
        self.__name=name
        self.__age=age
        self.__salary=salary

    def agregar_cuenta(self, cuenta):
        self.cuentas.append(cuenta)

    def get_cuentas(self):
        return self.cuentas

    def set_cuentas(self, cuentas):
        self.cuentas = cuentas

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name

    def get_age(self):
        return self.__age

    def set_age(self, age):
        self.__age = age

    def get_salary(self):
        return self.__salary

    def set_salary(self, salary):
        self.__salary = salary

    def get_account(self):
        return self.account

    def set_account(self, account):
        self.account = account

    def get_information(self):
        return f"El nombre del empleado es {self.name}, su cuenta {self.account.get_account_number()}, su saldo actual es {self.account.get_monto():.2f} y la cuenta es de tipo {self.account.tipo}"

    def ver_cuentas(self):
        print("\nINFORMACIÓN DEL EMPLEADO")
        print(f"Nombre: {self.__name} Edad: {self.get_age()} Salario: {self.__salary}")
        print("****Cuentas bancarias:")
        for b in self.cuentas:
                b.mostrar_informacion()

    def buscar_cuenta(self, numero_cuenta):
        for cuenta in self.cuentas:
            if cuenta.get_account_number() == numero_cuenta:
                return cuenta
        return None


    def mostrar_info(self):
        print(f"Nombre {self.__name}, edad: {self.__age}")

   

File: Tarea7_U2/src/tarea7_u2/Employee.java

package tarea7_u2;
import java.util.*;

public class Employee {
    private String name;
    private  int age;
    private double salary;
    BankAccount account; 
    List <BankAccount> cuentas = new ArrayList<>();

    public Employee(String name, int age, double salary, BankAccount account) {
        this.name = name;
        this.age = age;
        this.salary = salary;
        this.account = account;
    }
    
    
    
    public Employee(String name, int age,  char tipo, double salary, long account) {
        String formatAccoundType = Character.toString(tipo).toUpperCase();
        if( formatAccoundType.equals("A") || formatAccoundType.equals("B") || formatAccoundType.equals("C") ){
            this.name = name;
            this.age = age;
            this.salary = salary;
            this.account = new BankAccount (account, tipo);
        }
        else 
        {
            System.out.println("Ingresa un tipo de cuenta valido");
        }
            
    }

    public List<BankAccount> getCuentas() {
        return cuentas;
    }

    public void setCuentas(List<BankAccount> cuentas) {
        this.cuentas = cuentas;
    }
    
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public BankAccount getAccount() {
        return account;
    }

    public void setAccount(BankAccount account) {
        this.account = account;
    }
    
    
    public String getInformation (){
          
       return String.format("El nombre del empleado es  %s su cuenta  %d , su saldo actual es %.2f y la "
        + "cuenta es de tipo %s  ", this.name, this.account.getAccountNumber(), 
        this.account.getMonto(), this.account.tipo);
      
      }
    
    public void verCuentas() 
    {
        System.out.println("\nINFORMACIÓN DEL EMPLEADO");
        System.out.println("Nombre: " + this.getName() + "\nEdad: " + this.getAge() + "\nSalario: " + this.getSalary());
        System.out.println("****Cuentas bancarias:");
        for (BankAccount b : cuentas) {
            b.mostrarInformacion();
        }
    }
    public BankAccount buscarCuenta(long numeroCuenta) 
    {
        for (BankAccount cuenta : cuentas)
        {
            if (cuenta.getAccountNumber() == numeroCuenta) 
            {
                return cuenta;
            }
        }
        return null;
    }
}
